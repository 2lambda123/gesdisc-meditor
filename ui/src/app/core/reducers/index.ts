import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
