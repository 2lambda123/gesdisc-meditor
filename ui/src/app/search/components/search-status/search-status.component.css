.search-status-bar {
	margin-bottom: 30px;
	color: grey;
}

.add-new-button {
	margin-top: -10px;
	margin-left: 10px;
	width: 120px;
	background-color: #4286f4;
}

.sortby-select {
	margin-left: 10px;
}
