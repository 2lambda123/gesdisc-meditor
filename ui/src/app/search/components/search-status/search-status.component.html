<div fxLayout="row wrap" fxLayoutAlign="center start" class="search-status-bar">
	<div fxFlex="40">
		Showing {{ filteredCount > 0 ? 1 : 0 }} - {{ filteredCount }} of {{ resultCount }} {{modelName}} documents
	</div>
	<div fxFlex="4"></div>

	Sort by:
	<div fxFlex="8" class="sortby-select">
		<mat-select [(value)]="sortBy">
			<mat-option value="oldest">Oldest</mat-option>
    	<mat-option value="newest">Newest</mat-option>
		</mat-select>
	</div>
	<button mat-raised-button class="add-new-button" fxFlex="5">
		<mat-icon>add</mat-icon>Add new
		</button>
</div>
