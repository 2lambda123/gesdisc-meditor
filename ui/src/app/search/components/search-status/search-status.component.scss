.search-status-bar {
	display: flex;
	align-items: center;
	justify-content: space-between;
	margin-top: 40px;
	margin-bottom: 30px;
	color: grey;

	& > div {
		align-items: center;
		display: flex;
		flex: 1;
	}

	& > div:last-child {
		justify-content: flex-end;
	}
}

.add-new-button {
	margin-top: -10px;
	margin-left: 10px;
	width: 120px;
}

.add-new-button mat-icon {
	margin-right: 5px;
}

.search-select {
	display: flex;
	margin-left: 10px;
	margin-right: 10px;

	mat-label {
		margin-right: 5px;
		white-space: nowrap;
	}

	mat-select {
		min-width: 100px;
	}
}
