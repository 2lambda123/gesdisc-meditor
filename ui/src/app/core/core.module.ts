import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
