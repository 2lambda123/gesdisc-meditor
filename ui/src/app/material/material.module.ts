import { NgModule } from '@angular/core';

import {
  MatInputModule,
  MatCardModule,
  MatButtonModule,
  MatSidenavModule,
  MatListModule,
  MatIconModule,
  MatToolbarModule,
  MatProgressSpinnerModule,
  MatDatepickerModule,
  MatCheckboxModule,
  MatSelectModule
} from '@angular/material';

@NgModule({
  imports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
  exports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
})
export class MaterialModule {}
