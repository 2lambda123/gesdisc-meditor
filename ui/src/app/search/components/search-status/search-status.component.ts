import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core'
import { Edge } from '../../../service'

@Component({
    selector: 'med-search-status',
    templateUrl: './search-status.component.html',
    styleUrls: ['./search-status.component.css'],
})
export class SearchStatusComponent implements OnInit {
    @Input() filteredCount: number
    @Input() resultCount: number
    @Input() modelName: string
    @Input() actions: Edge[]
    @Input() privileges: string[]
    @Output() addNew = new EventEmitter<string>()
    @Output() sortByChanged = new EventEmitter()

    sortBy = 'desc'

    constructor() {}

    ngOnInit() {}
}
