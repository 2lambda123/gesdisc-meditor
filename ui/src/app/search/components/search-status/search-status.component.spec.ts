import { ComponentFixture, TestBed  } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { SearchStatusComponent } from './search-status.component';
import { MatSelectModule, MatButtonModule, MatIconModule } from '@angular/material';

describe('SearchStatusComponent', () => {
	let fixture: ComponentFixture<SearchStatusComponent>;
	let instance: SearchStatusComponent;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				NoopAnimationsModule,
				MatSelectModule,
				MatButtonModule,
				MatIconModule
			],
			declarations: [
				SearchStatusComponent
			]
		});

		fixture = TestBed.createComponent(SearchStatusComponent);
		instance = fixture.componentInstance;

	});

	describe('with no data', () => {
		it('should not have any initial input, should have sortBy set default to newest', () => {
			expect(instance.filteredCount).toBeUndefined();
			expect(instance.resultCount).toBeUndefined();
			expect(instance.modelName).toBeUndefined();
			expect(instance.sortBy).toEqual('newest');
		});
	});

	describe('with mock data', () => {

		it('should compile with minimum input', () => {
			instance.filteredCount = 10;
			instance.resultCount = 15;
			instance.modelName = 'test';
			instance.sortBy = 'oldest';
      instance.privileges = ['create'];
      instance.actions = [{
        label: 'Add new'
      }];
			fixture.detectChanges();
			expect(fixture).toMatchSnapshot();
      instance.privileges = [];
			fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
		});
	});
});
