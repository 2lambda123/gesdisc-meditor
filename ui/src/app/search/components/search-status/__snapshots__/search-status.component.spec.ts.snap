// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchStatusComponent with mock data should compile with minimum input 1`] = `
<med-search-status
  filteredCount={[Function Number]}
  modelName={[Function String]}
  resultCount={[Function Number]}
  sortBy={[Function String]}
>
  <div
    class="search-status-bar"
    fxlayout="row wrap"
    fxlayoutalign="center start"
  >
    <div
      fxflex="40"
    >
       Showing 1 - 10 of 15 test documents 
    </div>
    <div
      fxflex="4"
    />
     Sort by: 
    <div
      class="sortby-select"
      fxflex="8"
    >
      <mat-select
        aria-disabled="false"
        aria-invalid="false"
        aria-multiselectable="false"
        aria-required="false"
        class="mat-select ng-tns-c5-0"
        id="mat-select-1"
        ng-reflect-value="oldest"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-hidden="true"
          cdk-overlay-origin=""
          class="mat-select-trigger"
        >
          <div
            class="mat-select-value"
            ng-reflect-ng-switch="true"
          >
            
            <span
              class="mat-select-placeholder ng-tns-c5-0 ng-star-inserted"
            >
              Â 
            </span>
            
          </div>
          <div
            class="mat-select-arrow-wrapper"
          >
            <div
              class="mat-select-arrow"
            />
          </div>
        </div>
        
      </mat-select>
    </div>
    <button
      class="add-new-button mat-raised-button"
      fxflex="5"
      mat-raised-button=""
    >
      <span
        class="mat-button-wrapper"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon material-icons"
          role="img"
        >
          add
        </mat-icon>
        Add new 
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </div>
</med-search-status>
`;
